### ISSUE TITLE
A db API for handling annotations

### ISSUE THREAD

**Issue by GavinHuttley** [cogent3/cogent3#1195]
Consider an example. Using the `RichGenbankParser` to load a bacterial genome takes forever because the annotations are always converted into `_Annotatable` instances. There are too many objects created on the fly, better to only create these objects on demand.

As a step towards addressing this, I propose we implement an `AnnotationDbABC` (abstract base class) which defines two key methods.
- `find_records()` which takes a `seq_name`, `bio_type`, `identifier`, `start`, `end`, and `strand` arguments and returns a dictionary with the essential components for building a `_Annotatable`.
- `describe()` which takes a `seq_name`, `bio_type`, `identifier` arguments as booleans for whether to return the distinct values for these. This method should return a dict of `{<arg_name>: set(<results>), ...}`

The two major use-cases to solve concern annotations in GFF3 or GenBank formats. So the issue will use the existing `cogent3` parsers to load.

In summary:

- add new file `cogent3/core/annotation_db.py`
    - `annotation_db` contains the abstract base class and 2 subclasses, one for GenBank and one for GFF3. The latter two should use the built in sqlite db engine.
- add new file `tests/test_core/test_annotation_db.py`
    - tests of the above

**Comment by kiratalreja3** [comment]
Hey @GavinHuttley , I would like to take this up!

**Comment by GavinHuttley** [comment]
fixed in e330639
